ts/Data/GoogleDataStore.ts

```diff
@@ -0,0 +1,227 @@
+
+            // If both row and col falls inside start and end set the
+            // transposed cell value in the newly created columns
+            if (gc >= startColumn && gc <= endColumn &&
+                gr >= startRow && gr <= endRow) {
+
+                cellInner = cell.gs$cell || cell.content;
+
+                val = null;
+
+                if (cellInner.numericValue) {
+                    if (cellInner.$t.indexOf('/') >= 0 ||
+                        cellInner.$t.indexOf('-') >= 0) {
+                        // This is a date - for future reference.
+                        val = cellInner.$t;
+                    } else if (cellInner.$t.indexOf('%') > 0) {
+                        // Percentage
+                        val = parseFloat(cellInner.numericValue) * 100;
+                    } else {
+                        val = parseFloat(cellInner.numericValue);
+                    }
+                } else if (cellInner.$t && cellInner.$t.length) {
+                    val = cellInner.$t;
+                }
+
+                columns[gc - startColumn][gr - startRow] = val;
+            }
+        }
+
+        // Insert null for empty spreadsheet cells (#5298)
+        columns.forEach(function (
+            column: Array<Highcharts.DataValueType>
+        ): void {
+            for (i = 0; i < column.length; i++) {
+                if (typeof column[i] === 'undefined') {
+                    column[i] = null as any;
+                }
+            }
+        });
+
+        /* *
+         * TODO:
+         * add data to dataTable
+         * ...
+         *
+         * */
+    }
+
+    private fetchSheet(): (boolean) {
+        const parseSheet = this.parseSheet;
+        const fetchSheet = this.fetchSheet;
+
+        const enablePolling = this.enablePolling;
+        const dataRefreshRate = this.dataRefreshRate;
+
+        const url = [
+            'https://spreadsheets.google.com/feeds/cells',
+            this.googleSpreadsheetKey,
+            this.worksheet,
+            'public/values?alt=json'
+        ].join('/');
+
+        ajax({
+            url: url,
+            dataType: 'json',
+            success: function (json: Highcharts.JSONType): void {
+
+                parseSheet(json);
+
+                // Polling
+                if (enablePolling) {
+                    setTimeout(
+                        function (): void {
+                            fetchSheet();
+                        },
+                        dataRefreshRate * 1000
+                    );
+                }
+            },
+            error: function (
+                xhr: XMLHttpRequest,
+                text: (string|Error)
+            ): void {
+                /* *
+                 * TODO:
+                 * catch error
+                 * ...
+                 *
+                 * */
+                // console.log(text);
+            }
+        });
+
+        return true;
+    }
+
+    public load(): boolean {
```

Comment:
The return type of `load` should be void, so that we later can replace it with a promise, when a refactoring is possible.

Reply:
Done